File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(n: usize) usize {
    return (n * (1 + n) / 2) * (n * (1 + n) / 2);
}

pub fn sumOfSquares(n: usize) usize {
    return (n * (n + 1) * (2 * n + 1)) / 6;
}

pub fn differenceOfSquares(n: usize) usize {
    return squareOfSum(n) - sumOfSquares(n);
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	return (n * (1 + n) / 2) * (n * (1 + n) / 2)
	// (n * (1 + n) / 2).pow(2)
}

func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
